src/ext/php7/dispatch.c

```diff
@@ -0,0 +1,136 @@
+#include "dispatch.h"
+
+#include <Zend/zend_exceptions.h>
+#include <php.h>
+
+#include <ext/spl/spl_exceptions.h>
+
+#include "compatibility.h"
+#include "ddtrace.h"
+#include "debug.h"
+
+ZEND_EXTERN_MODULE_GLOBALS(ddtrace)
+
+static int ddtrace_is_all_lower(zend_string *s) {
+    unsigned char *c, *e;
+
+    c = (unsigned char *)ZSTR_VAL(s);
+    e = c + ZSTR_LEN(s);
+
+    int rv = 1;
+    while (c < e) {
+        if (isupper(*c)) {
+            rv = 0;
+            break;
+        }
+        c++;
+    }
+    return rv;
+}
+
+zend_function *ddtrace_function_get(const HashTable *table, zval *name) {
+    if (Z_TYPE_P(name) != IS_STRING) {
+        return NULL;
+    }
+
+    zend_string *to_free = NULL, *key = Z_STR_P(name);
+    // todo: see if this can just be replaced with zend_string_tolower, which already does an optimization like this
```

Comment:
Is this for another PR or was this intended to be included in this one?

Reply:
I didn't want to risk a behavior change here, since this was meant to be purely refactoring.